apis/networking/v1beta1/apiproduct_types.go

```diff
@@ -89,11 +89,26 @@ func (a *APISelector) APINamespacedName() types.NamespacedName {
 	return types.NamespacedName{Namespace: a.Namespace, Name: name}
 }
 
-type InfraRateLimitSpec struct {
+type AuthRateLimitSpec struct {
 	MaxValue int32 `json:"max_value"`
 	Period   int32 `json:"period_in_seconds"`
 }
 
+type PreAuthRateLimitSpec struct {
+	MaxValue int32 `json:"max_value"`
+	Period   int32 `json:"period_in_seconds"`
+}
+
+type RateLimitSpec struct {
+	// PreAuthRateLimitSpec configures rate limit parameters per remote address
+	// +optional
+	PreAuthRateLimit *PreAuthRateLimitSpec `json:"preAuthenticated,omitempty"`
```

Comment:
I think we should call it something else. "PreAuthRateLimit" makes sense if know that it opposes to "AuthenticatedRateLimit". If the user is planning on using "auth", speaking of "pre-auth" sounds weird IMO.

Maybe we should call this one just "RateLimit"?

I also like the idea (once suggested by @jmprusi) of defining "phases".